src/Io/SapiHandler.php

```diff
@@ -108,6 +108,10 @@ public function sendResponse(ResponseInterface $response): void
         ini_set('default_charset', '');
         foreach ($response->getHeaders() as $name => $values) {
             foreach ($values as $value) {
+                if ($name === 'Set-Cookie') {
+                    header($name . ': ' . $value, false);
+                    continue;
+                }
```

Comment:
I'm wondering if this `if` is really necessary here. Using it would imply there are two ways of going for this, of which only one is currently tested (if you remove the `if` all tests are still green, right?)

There's also a problem with the `$name === 'Set-Cookie'` since HTTP header keys are defined to be case-insensitive.

So my question would be, is this `if` relevant here?

Reply:
Good point about the headers being case-insensitive, missed that when adding the check.

Initially I didn't have the check at all and I also think that just adding `false` to `header()` should be sufficient but that also means that other headers could be duplicated which currently wasn't the case.

The main reason I added the check was that 12 tests failed without it, at the moment I'm not entirely sure why, will look into that later.

Reply:
Made the fix simpler again, as proposed, but either way I had to change the tests a little bit as if the headers are not overwritten, they'd add up to quite a lot in `xdebug_get_headers()` making the current solution quite bulky and prone to errors. Also, I could not get one test to pass, it would pass on its own, however. Now it works all the time.

Reply:
Nice :+1: